src/mpid/ch4/shm/ipc/xpmem/xpmem_mem.h

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (C) by Argonne National Laboratory
+ *     See COPYRIGHT in top-level directory
+ */
+#ifndef XPMEM_MEM_H_INCLUDED
+#define XPMEM_MEM_H_INCLUDED
```

Comment:
We usually use `_pre.h` for type definitions and `_post.h` for function declarations.  I'd recommend the same convention, so it's clear what API is being exposed.